fix test flake

https://github.com/factor-finance/xusd-contracts/actions/runs/1737803182/attempts/1
```
1) Vault Redeem
       Should have correct balances on consecutive mint and redeem:

      AssertionError: Expected "100000000000000000000" to be greater than or equal 120988790100000000000
      + expected - actual

       {
      -  "_hex": "0x068f0e8791e2cc4800"
      +  "_hex": "0x056bc75e2d63100000"
         "_isBigNumber": true
       }
      
      at Proxy.<anonymous> (test/helpers.js:11:32)
      at Proxy.methodWrapper (node_modules/chai/lib/chai/utils/addMethod.js:57:25)
      at Proxy.<anonymous> (test/helpers.js:22:28)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at runNextTicks (internal/process/task_queues.js:66:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at Context.<anonymous> (test/vault/redeem.js:310:11)
```

--- a/test/vault/redeem.js
+++ b/test/vault/redeem.js
@@ -305,8 +305,12 @@ describe("Vault Redeem", function () {
     for (const [user, startBalance] of usersWithBalances) {
       for (const [asset, units] of assetsWithUnits) {
         for (const amount of [5.09, 10.32, 20.99, 100.01]) {
-          asset.connect(user).approve(vault.address, units(amount.toString()));
-          vault.connect(user).mint(asset.address, units(amount.toString()), 0);
+          await asset
+            .connect(user)
+            .approve(vault.address, units(amount.toString()));
+          await vault
+            .connect(user)
+            .mint(asset.address, units(amount.toString()), 0);
           await expect(user).has.an.approxBalanceOf(
             (startBalance + amount).toString(),
             xusd
@@ -415,10 +419,10 @@ describe("Vault Redeem", function () {
 
     await vault.connect(anna).redeemAll(0);
 
-    dai.connect(josh).approve(vault.address, daiUnits("50"));
-    vault.connect(josh).mint(dai.address, daiUnits("50"), 0);
-    dai.connect(matt).approve(vault.address, daiUnits("100"));
-    vault.connect(matt).mint(dai.address, daiUnits("100"), 0);
+    await dai.connect(josh).approve(vault.address, daiUnits("50"));
+    await vault.connect(josh).mint(dai.address, daiUnits("50"), 0);
+    await dai.connect(matt).approve(vault.address, daiUnits("100"));
+    await vault.connect(matt).mint(dai.address, daiUnits("100"), 0);
 
     let newBalance = await usdc.balanceOf(await anna.getAddress());
     let newDaiBalance = await dai.balanceOf(await anna.getAddress());

